### DISCUSSION TITLE
appearance panel search shown as not floating pane

### DISCUSSION THREAD

**Discussion by jeusdi** [mrjones2014/legendary.nvim#257] [Q&A] [answered]
I've just installed `mrjones2014/legendary.nvim`.

It seems to work well, however I'm not getting an appearance behavior similar to floating window panel to search. Here you can see, what I mean into this [asciinema](https://asciinema.org/a/hXvOL4ICtmrErSgAmAX7IyHeh).

I was expecting to get a floating panel like shown into [landing github project page](https://github.com/mrjones2014/legendary.nvim).

My current configuration is [here](https://gitlab.com/jeusdit/dotfiles/-/blob/main/lua/keymappings.lua#L100).

Any ideas?

**Comment by mrjones2014** [comment, marked as answer]
See https://github.com/mrjones2014/legendary.nvim#prerequisites

Also, for the which-key auto-register functionality to work, looking at your config, you'd need to rearrange the lines like this:

```lua
require('legendary').setup({ which_key = { auto_register = true } })
wk.register(mappings, opts)
```

Rather than calling `wk.register` first.

**Reply by jeusdi** [reply]
Thanks @mrjones2014 .

I already read above suggested [prerequisites](https://github.com/mrjones2014/legendary.nvim#prerequisites) section but I have not quite understood it at all.
Does it means that I need to have installed both `telescope` and `dressing`? Or only one of them? Currently I've installed `telescope` (nvim-telescope/telescope.nvim). I also need `dressing` plugin?

**Reply by mrjones2014** [reply]
`telescope.nvim` provides the fuzzy finder. `dressing.nvim` hooks it up to `vim.ui.select()` which is what `legendary.nvim` uses internally.

So to mimic my setup you will need both of those plugins, however there are also some alternatives like [fzf](https://github.com/junegunn/fzf.vim). But, you will still need something that hooks them up to `vim.ui.select()`. For that I'd recommend still using `dressing.nvim`. It supports wiring up `vim.ui.select()` with `telescope.nvim`, `fzf.vim`, `nui.nvim` and a default built-in select backend.
